aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -161,10 +161,76 @@ static void where_kernel_impl(TensorIterator &iter, ScalarType condition_type) {
   });
 }
 
+static void isposinf_kernel_for_bool_impl(TensorIterator& iter) {
+  if (c10::isIntegralType(iter.input_dtype(), /*include_bool=*/true)) {
```

Comment:
See recommendation above for reusing the same trick that `isfinite` does and not needing to dispatch on integral types.

Reply:
addressed